src/xpk/commands/shell.py

```diff
@@ -0,0 +1,122 @@
+"""
+
+from ..core.commands import run_command_with_full_controls, run_command_for_value, run_command_with_updates
+from ..utils.console import xpk_exit, xpk_print
+from argparse import Namespace
+
+from ..core.kjob import AppProfileDefaults, PodTemplateDefaults
+
+
+def shell(args: Namespace):
+  """Enter interactive shell.
+  Args:
+    args: user provided arguments for running the command.
+  Returns:
+    0 if successful and 1 otherwise.
+  """
+  exisitng_shell_pod_name = get_existing_shell_pod_name(args)
+
+  if exisitng_shell_pod_name is None:
+    return_code = connect_to_new_interactive_shell(args)
+  else:
+    return_code = connect_to_existing_interactive_shell(
+        exisitng_shell_pod_name, args
+    )
+
+  if return_code != 0:
+    xpk_print(f'The command failed with code {return_code}.')
+    xpk_exit(return_code)
+
+  xpk_exit(0)
+
+
+def get_existing_shell_pod_name(args: Namespace) -> str | None:
+  return_code, shell_name = run_command_for_value(
+      command=(
+          'kubectl get pods --no-headers --field-selector status.phase=Running'
+          ' -o custom-columns=":metadata.name"'
+      ),
+      task='Get existing interactive shell pod name.',
+      global_args=args,
+  )
+  if return_code != 0:
+    xpk_print(
+        f'Encounter an error with a code {return_code} when checking for'
+        ' existing running shell.'
+    )
+    xpk_exit(return_code)
+
+  pod_names = shell_name.strip().split('\n')
+  kjob_pod_names = [
+      name for name in pod_names if AppProfileDefaults.NAME.value in name
+  ]
+  shell_pod_names = [name for name in kjob_pod_names if 'interactive' in name]
+
+  return shell_pod_names[0] if shell_pod_names else None
+
+
+exit_instructions = 'To exit the shell input "exit".'
+
+
+def connect_to_new_interactive_shell(args: Namespace) -> int:
+  return run_command_with_full_controls(
+      command=(
+          'kubectl-kjob create interactive --profile'
+          f' {AppProfileDefaults.NAME.value} --pod-running-timeout 30s'
+      ),
+      task='Creating new interactive shell and entering it',
+      global_args=args,
+      instructions=exit_instructions,
+  )
+
+
+def connect_to_existing_interactive_shell(
+    pod_name: str, args: Namespace
+) -> int:
+  return run_command_with_full_controls(
+      command=(
+          f'kubectl exec --stdin --tty {pod_name} --'
+          f' {PodTemplateDefaults.INTERACTIVE_COMMAND.value}'
+      ),
+      task='Entering existing interactive shell',
+      global_args=args,
+      instructions=exit_instructions,
+  )
+
+
+def shell_stop(args: Namespace):
+  """Stop the running interactive shell.
+  Args:
+    args: user provided arguments for running the command.
+  Returns:
+    0 if successful and 1 otherwise.
+  """
+  exisitng_shell_pod_name = get_existing_shell_pod_name(args)
+
+  if exisitng_shell_pod_name is None:
+    xpk_print('There is no shell running to stop')
+    xpk_exit(0)
+
+  return_code = run_command_with_updates(
```

Comment:
Are we sure we should delete shell on stop command? We can't run it again? Can this shell persist data between user sessions?

Reply:
ok, but function is named shell stop and the docstring says stop the running interactive shell. Should't it have delete mentioned in name or docstring?